cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -106,40 +108,52 @@ protected final void stopScheduler() throws SchedulerException {
 
   @Override
   public void addProgramSchedule(ProgramSchedule schedule) throws AlreadyExistsException, SchedulerException {
-    if (schedule.getTrigger() instanceof TimeTrigger) {
+    if (callTimeScheduler(schedule)) {
       timeScheduler.addProgramSchedule(schedule);
-    } else if (schedule.getTrigger() instanceof StreamSizeTrigger) {
+    }
+    if (callStreamSizeScheduler(schedule)) {
       streamSizeScheduler.addProgramSchedule(schedule);
     }
   }
 
   @Override
   public void deleteProgramSchedule(ProgramSchedule schedule) throws NotFoundException, SchedulerException {
-    if (schedule.getTrigger() instanceof TimeTrigger) {
+    if (callTimeScheduler(schedule)) {
       timeScheduler.deleteProgramSchedule(schedule);
-    } else if (schedule.getTrigger() instanceof StreamSizeTrigger) {
+    }
+    if (callStreamSizeScheduler(schedule)) {
       streamSizeScheduler.deleteProgramSchedule(schedule);
     }
   }
 
   @Override
   public void suspendProgramSchedule(ProgramSchedule schedule) throws NotFoundException, SchedulerException {
-    if (schedule.getTrigger() instanceof TimeTrigger) {
+    if (callTimeScheduler(schedule)) {
       timeScheduler.suspendProgramSchedule(schedule);
-    } else if (schedule.getTrigger() instanceof StreamSizeTrigger) {
+    }
+    if (callStreamSizeScheduler(schedule)) {
       streamSizeScheduler.suspendProgramSchedule(schedule);
     }
   }
 
   @Override
   public void resumeProgramSchedule(ProgramSchedule schedule) throws NotFoundException, SchedulerException {
-    if (schedule.getTrigger() instanceof TimeTrigger) {
+    if (callTimeScheduler(schedule)) {
       timeScheduler.resumeProgramSchedule(schedule);
-    } else if (schedule.getTrigger() instanceof StreamSizeTrigger) {
+    }
+    if (callStreamSizeScheduler(schedule)) {
       streamSizeScheduler.resumeProgramSchedule(schedule);
     }
   }
 
+  private boolean callTimeScheduler(ProgramSchedule schedule) {
+    return schedule.getTrigger() instanceof TimeTrigger || schedule.getTrigger() instanceof AbstractCompositeTrigger;
```

Comment:
because we don't want to include `StreamSizeTrigger` in new API any more (like in `ScheduleBuilder`), so `StreamSizeTrigger` won't be contained in any composite triggers. However, composite trigger can contain `TimeTrigger`